bindgen/private/bindgen.bzl

```diff
@@ -281,33 +284,23 @@ def _rust_bindgen_impl(ctx):
         tools = tools,
     )
 
-    if ctx.attr.leak_symbols:
-        # buildifier: disable=print
-        print("WARN: rust_bindgen.leak_symbols is set to True for {} - please file an issue at https://github.com/bazelbuild/rules_rust/issues explaining why this was necessary, as this support will be removed soon.".format(ctx.label))
-        providers = [cc_common.merge_cc_infos(
-            direct_cc_infos = [cc_lib[CcInfo]],
-        )]
-    else:
-        providers = [
-            _generate_cc_link_build_info(ctx, cc_lib),
-            # As in https://github.com/bazelbuild/rules_rust/pull/2361, we want
-            # to link cc_lib to the direct parent (rlib) using `-lstatic=<cc_lib>` rustc flag
-            # Hence, we do not need to provide the whole CcInfo of cc_lib because
-            # it will cause the downstream binary to link the cc_lib again
-            # (same effect as setting `leak_symbols` attribute above)
-            # The CcInfo here only contains the custom link flags (i.e. linkopts attribute)
-            # specified by users in cc_lib
-            CcInfo(
-                linking_context = cc_common.create_linking_context(
-                    linker_inputs = depset([cc_common.create_linker_input(
-                        owner = ctx.label,
-                        user_link_flags = _get_user_link_flags(cc_lib),
-                    )]),
-                ),
+    return [
+        _generate_cc_link_build_info(ctx, cc_lib),
+        # As in https://github.com/bazelbuild/rules_rust/pull/2361, we want
+        # to link cc_lib to the direct parent (rlib) using `-lstatic=<cc_lib>` rustc flag
+        # Hence, we do not need to provide the whole CcInfo of cc_lib because
+        # it will cause the downstream binary to link the cc_lib again
+        # (same effect as setting `leak_symbols` attribute above)
```

Comment:
nit: remove this reference to `leak_symbols`?

Reply:
Good catch! 🙏 

Updated this comment and did some extra cleanup